src/libraries/System.Memory/tests/SequenceReader/Rewind.cs

```diff
@@ -120,5 +120,20 @@ public void RewindEmptyFirstSpan()
             reader.Rewind(2);
             Assert.Equal(new byte[] { 1, 2 }, reader.CurrentSpan.ToArray());
         }
+
+        [Fact]
+        public void RewindZero_DoesNothing()
+        {
+            var emptySeq = ReadOnlySequence<byte>.Empty;
+            var reader = new SequenceReader<byte>(emptySeq);
+
+            Assert.True(reader.End);
+            Assert.False(reader.TryRead(out byte _));
+
+            reader.Rewind(0);
+
+            Assert.True(reader.End);
+            Assert.False(reader.TryRead(out byte _));
```

Comment:
`Rewind(0)` set _moreData to true and all other methods thought that's still not the end